pkgs/development/libraries/opencv/4.x.nix

```diff
@@ -0,0 +1,287 @@
+{ lib, stdenv
+, fetchurl, fetchFromGitHub
+, cmake, pkgconfig, unzip, zlib, pcre, hdf5
+, glog, boost, google-gflags, protobuf
+, config
+
+, enableJPEG      ? true, libjpeg
+, enablePNG       ? true, libpng
+, enableTIFF      ? true, libtiff
+, enableWebP      ? true, libwebp
+, enableEXR ? (!stdenv.isDarwin), openexr, ilmbase
+, enableJPEG2K    ? true, jasper
+, enableEigen     ? true, eigen
+, enableOpenblas  ? true, openblas
+, enableContrib   ? true
+
+, enableCuda      ? (config.cudaSupport or false), cudatoolkit
+
+, enableUnfree    ? false
+, enableIpp       ? false
+, enablePython    ? false, pythonPackages
```

Comment:
Admittedly I'm not sure if that's a good idea. Python seems to be used often for such cases today and a full rebuild takes a certain amount of power and time and using Python is also one of the advertised ways of using `opencv4` according the documentation.

Also, how is this supposed to be used? I'd love to see it possible to simply add `opencv` to a python environment to start hacking (but there's no build product in `lib/pythonX.X`).

Reply:
@Ma27 thanks for reminding me to also have Python support. I forced-pushed a [`opencv4`](https://github.com/NixOS/nixpkgs/pull/51027/files#diff-2afbba83e1b14f9e0c304c769831a0e2R1294) attribute in `python-packages.nix`. Note that `4.0.0.nix` is a copy of `3.0.0.nix` so the way we handle Python is the same as with `opencv3`.